src/vm/rejit.cpp

```diff
@@ -685,12 +729,210 @@ HRESULT ReJitManager::UpdateActiveILVersions(
+        ILCodeVersion* pILCodeVersion = pCodeActivationBatch->m_methodsToActivate.Append();
+        if (pILCodeVersion == NULL)
+        {
+            return E_OUTOFMEMORY;
+        }
+        if (fIsRevert)
+        {
+            // activate the original version
+            *pILCodeVersion = ILCodeVersion(pModule, methodDef);
+        }
+        else
+        {
+            // activate an unused or new IL version
+            hr = ReJitManager::BindILVersion(pCodeVersionManager, pModule, methodDef, pILCodeVersion, flags);
+            if (FAILED(hr))
+            {
+                _ASSERTE(hr == E_OUTOFMEMORY);
+                return hr;
+            }
+        }
+    }
+
+    return hr;
+}
+
+// static
+HRESULT ReJitManager::UpdateNativeInlinerActiveILVersions(
+    SHash<CodeActivationBatchTraits>   *pMgrToCodeActivationBatch,
+    MethodDesc                         *pInlinee,
+    BOOL                                fIsRevert,
+    COR_PRF_REJIT_FLAGS                 flags)
+{
+    _ASSERTE(pMgrToCodeActivationBatch != NULL);
+    _ASSERTE(pInlinee != NULL);
+    
+    HRESULT hr = S_OK;
+
+    // Iterate through all modules, for any that are NGEN or R2R need to check if there are inliners there and call 
+    // RequestReJIT on them
+    // TODO: is the default domain enough for coreclr?
+    AppDomain::AssemblyIterator domainAssemblyIterator = SystemDomain::System()->DefaultDomain()->IterateAssembliesEx((AssemblyIterationFlags) (kIncludeLoaded));
+    CollectibleAssemblyHolder<DomainAssembly *> pDomainAssembly;
+    NativeImageInliningIterator inlinerIter;
+    while (domainAssemblyIterator.Next(pDomainAssembly.This()))
+    {
+        _ASSERTE(pDomainAssembly != NULL);
+        _ASSERTE(pDomainAssembly->GetAssembly() != NULL);
+
+        DomainModuleIterator domainModuleIterator = pDomainAssembly->IterateModules(kModIterIncludeLoaded);
+        while (domainModuleIterator.Next())
+        {
+            Module * pCurModule = domainModuleIterator.GetModule();
+            if (pCurModule->HasNativeOrReadyToRunInlineTrackingMap())
+            {
+                inlinerIter.Reset(pCurModule, pInlinee);
+
+                MethodDesc *pInliner = NULL;
+                while (inlinerIter.Next())
+                {
+                    pInliner = inlinerIter.GetMethodDesc();
+                    {
+                        CodeVersionManager *pCodeVersionManager = pCurModule->GetCodeVersionManager();
+                        CodeVersionManager::TableLockHolder lock(pCodeVersionManager);
+                        ILCodeVersion ilVersion = pCodeVersionManager->GetActiveILCodeVersion(pInliner);
+                        if (!ilVersion.HasDefaultIL())
+                        {
+                            // This method has already been ReJITted, no need to request another ReJIT at this point.
+                            // The ReJITted method will be in the JIT inliner check below.
+                            continue;
```

Comment:
Apologies if I have just paged this out but what happens when A inlines B by default and a profiler wants to do:
Edit function A
Edit function B
Revert the edit to A

If the profiler calls RejitWithInliner A, RejitWithInliner B, Revert A then it appears they are going to accidentally lose their edit to B that is now inlined in A again. Did we say they should use ReJIT instead of Revert and pass the original IL? 

It would probably help to write up the expected usage in a little design doc/documentation because it feels easy to accidentally do the wrong thing.

Reply:
I don't remember talking about this specific detail before. You are correct that the way they are currently implemented reverts will ignore any modified IL. 

I think the right course is documenting it and suggesting that the right behavior is either:
1. You only care about rejitting one method and revert will do what you want
2. You care about rejitting more than one method, and revert is dangerous, so RequestReJIT with the original IL

We could also change the current revert behavior from activating the original native code to effectively doing another rejit with the original IL, but I am not convinced it's worth it